rfcs/0112-demote-darwin-support-tier.md

```diff
@@ -0,0 +1,91 @@
+    - Due to Hydra being understaffed, this is a problem (see above).
+
+We are not only lacking CI machines, but also maintainers that can dedicate their time to work on fixing broken Darwin packages.
+
+## Exemplary collection of issues with Darwin
+
+The issues that are constantly encountered in relation to the Darwin platform have a genuine negative impact on the `nixpkgs` development experience. People are forced to deal with these problems even though they have no hardware to test on, lack the proper expertise, and are reluctant when it comes to investing additional time and energy to patch for the failing platform. Demoting Darwin to Tier 3 will allow more developers to refocus their attention on better supported platforms, care less about Darwin issues, and push that burden onto the Darwin maintainers.
+
+An excerpt of these issues are:
+
+- The only way to debug issues without owning a Mac is to run Darwin in a VM, of which the legal situation is unclear.
+   - Those licensing requirements are causing friction with our CI setup as well.
+- The sandboxing capabilities on MacOS are limited. This results in a special category of test failures only relevant to Darwin. It also prevents us from providing a community builder, because it could not be shared between multiple people that cannot trust each other.
+- The MacOS SDK cannot be updated and is stuck on 10.12 (released 2016/09) because Apple does not publish the required sources.
+    - https://github.com/NixOS/nixpkgs/issues/101229#issuecomment-938747052
+    - It is highly unlikely that we can report bugs to projects with such an outdated and unsupported SDK version
+- The MacOS SDK being outdated blocks nixpkgs moving to go 1.17:
+    - https://github.com/NixOS/nixpkgs/pull/127519#issuecomment-864926149
+    - This will become a growing pain point once more packages become go 1.17-only (tailscale, talosctl, …)
+- Multiple Packages were marked as broken, because they require symbols from newer MacOS SDK versions.
+    - https://github.com/NixOS/nixpkgs/commit/3ceb5ab5ed0d1fcbe53ef00621f44d61dc524796
+    - https://github.com/NixOS/nixpkgs/commit/c9a3ac5d3cb5e910238d01e534d74d5d50e4b6b7
+- Curl added a SystemConfiguration dependency for NAT64 support, which introduced a reference loop, requiring a downstream patch to workaround.
+    - This blocked updating to more recent curl versions for most of the last release cycle.
+    - https://github.com/NixOS/nixpkgs/pull/124502#issuecomment-850834981
+- Enabling brotli support by default in curl broke the Darwin stdenv, which draws in a great number of packages.
+    - https://github.com/NixOS/nixpkgs/pull/112947
+    - https://github.com/NixOS/nixpkgs/pull/115498
+
+# Detailed design
+[design]: #detailed-design
+
+- Together with this RFC, `0046-platform-support-tiers.md` will be updated accordingly (simply move `x86_64-darwin` down one section).
+- Whatever needs to be done to make Darwin not block any channels anymore (TODO).
```

Comment:
About arch-specific channels: I wrote this sentence in #46 badly, sorry. The context is impact on the other development, so there is no justification in RFC to restrict  configuration of channels not used for any higher-tier architecture.

About nixpkgs-unstable: I think there are cases where some annoying issues with boot process break NixOS tests and some non-NixOS Nix-on-GNU/Linux users switch to nixpkgs channels? Unsure, as I usually use git checkouts.

Reply:
How could that line be improved then?

Reply:
The line in the RFC? Maybe «No jobs on Hydra block any channel also used by a higher-tier platform.» would be an improvement and should not get too many objections. But maybe it can be improved even further.